pkg/scheduler/internal/cache/cache_test.go

```diff
@@ -1539,6 +1539,180 @@ func compareCacheWithNodeInfoSnapshot(cache *schedulerCache, snapshot *Snapshot)
+				Name: "node-0",
+				Labels: map[string]string{
+					v1.LabelZoneRegion:        "region-1",
+					v1.LabelZoneFailureDomain: "zone-1",
+				},
+			},
+		},
+		// Node 1: a node with proper region and zone labels in a different region and zone
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-1",
+				Labels: map[string]string{
+					v1.LabelZoneRegion:        "region-1",
+					v1.LabelZoneFailureDomain: "zone-1",
+				},
+			},
+		},
+		// Node 2: a node with proper region and zone labels in a region and zone
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-2",
+				Labels: map[string]string{
+					v1.LabelZoneRegion:        "region-2",
+					v1.LabelZoneFailureDomain: "zone-2",
+				},
+			},
+		},
+		// Node 3: a node with proper region and zone labels in a region and zone
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-3",
+				Labels: map[string]string{
+					v1.LabelZoneRegion:        "region-2",
+					v1.LabelZoneFailureDomain: "zone-2",
+				},
+			},
+		},
+		// Node 4: a node with proper region and zone labels in a region and zone
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-4",
+				Labels: map[string]string{
+					v1.LabelZoneRegion:        "region-2",
+					v1.LabelZoneFailureDomain: "zone-2",
+				},
+			},
+		},
+		// Node 5: a node with proper region and zone labels in a region and zone
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-5",
+				Labels: map[string]string{
+					v1.LabelZoneRegion:        "region-2",
+					v1.LabelZoneFailureDomain: "zone-2",
+				},
+			},
+		},
+		// Node 6: a node with proper region and zone labels in a region and zone
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-6",
+				Labels: map[string]string{
+					v1.LabelZoneRegion:        "region-2",
+					v1.LabelZoneFailureDomain: "zone-2",
+				},
+			},
+		},
+		// Node 7: a node with proper region and zone labels in a region and zone
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-7",
+				Labels: map[string]string{
+					v1.LabelZoneRegion:        "region-2",
+					v1.LabelZoneFailureDomain: "zone-2",
+				},
+			},
+		},
+	}
+
+	var cache *schedulerCache
+	var snapshot *Snapshot
+	type operation = func()
+
+	addNode := func(i int) operation {
+		return func() {
+			if err := cache.AddNode(nodes[i]); err != nil {
+				t.Error(err)
+			}
+			_, ok := snapshot.nodeInfoMap[nodes[i].Name]
+			if !ok {
+				snapshot.nodeInfoMap[nodes[i].Name] = cache.nodes[nodes[i].Name].info
+			}
+		}
+	}
+
+	updateSnapshot := func() operation {
+		return func() {
+			cache.updateNodeInfoSnapshotList(snapshot, true)
+			if err := compareCacheWithNodeInfoSnapshot(cache, snapshot); err != nil {
+				t.Error(err)
+			}
+		}
+	}
+
+	tests := []struct {
+		name       string
+		operations []operation
+		expected   []string
+	}{
+		{
+			name:       "Empty cache",
+			operations: []operation{},
+			expected:   []string{},
+		},
+		{
+			name:       "Single node",
+			operations: []operation{addNode(0)},
```

Comment:
nit: I know that you're trying to iterate each "operation", but I feel the iteration is not needed. If we simply wrap all the operations in a single `func()` here, L1699~L1671 ca be replaced with `test.f()`.